### ISSUE TITLE
aad integration does not work on 1.17

### ISSUE THREAD

**Issue by karataliu** [Azure/aks-engine#3042]
**Describe the issue**
AAD integration: https://github.com/Azure/aks-engine/blob/master/docs/topics/aad.md

On 1.16, works well
```
I0408 13:28:25.948048       1 flags.go:33] FLAG: --oidc-username-prefix=""
I0408 13:28:28.201904       1 oidc.go:290] OIDC: No x509 certificates provided, will use host's root CA set
```
On 1.17:
```
I0408 13:27:25.138731       1 flags.go:33] FLAG: --oidc-username-claim="oid"
I0408 13:27:25.138743       1 flags.go:33] FLAG: --oidc-username-prefix=""
I0408 13:27:25.791781       1 oidc.go:290] OIDC: No x509 certificates provided, will use host's root CA set
E0408 13:27:35.803722       1 oidc.go:232] oidc authenticator: initializing plugin: Get https://sts.windows.net/*/.well-known/openid-configuration: x509: certificate signed by unknown authority
E0408 13:27:45.872582       1 oidc.go:232] oidc authenticator: initializing plugin: Get https://sts.windows.net/*/.well-known/openid-configuration: x509: certificate signed by unknown authority
E0408 13:27:55.849293       1 oidc.go:232] oidc authenticator: initializing plugin: Get https://sts.windows.net/*/.well-known/openid-configuration: x509: certificate signed by unknown authority
```

oidc could not fetch the open id configuration due to ssl validation issue.

**AKS Engine Version***

**Kubernetes Version**

**Gatekeeper Version**

**To Reproduce**

**Expected behavior**

**Screenshots**

**Additional context**

**Please don't remove the text below**
Notify @az-policy-kube

**Comment by satishsudini** [comment]
We are facing same issue with the aks-engine v47.0 and kubernetes 1.17 seeing below error (error: You must be logged in to the server (Unauthorized))

**Comment by mhyllander** [comment]
I see the same with aks-engine v0.49.0 and kubernetes 1.17 and 1.18.

**Comment by jackfrancis** [comment]
@ritazh @sozercan are we aware of this? (I'm not)

**Comment by ericsuhong** [comment]
I am actually facing this issue (**error: You must be logged in to the server (Unauthorized)**) in kubernetes 1.16.9 as well, using aks-engine 0.50.0.

**Comment by ericsuhong** [comment]
We did not change anything related to AAD integration code, so I believe some regression was introduced somewhere either by aks-engine or Kubernetes releases.

**Comment by jackfrancis** [comment]
@ericsuhong are you saying that from the AAD platform perspective?

**Comment by DavidParks8** [comment]
We found how to resolve the issue. Eric will explain.

**Comment by ericsuhong** [comment]
@jackfrancis Turns out that this was a kubectl binary issue. I upgraded kubectl binary from 1.16-something (which was referenced by default by docker-desktop /bin folder in Windows) to 1.18, and now everything works!

**Comment by DavidParks8** [comment]
@karataliu can you please try with the latest kubectl and report back on if it works?

**Comment by jackfrancis** [comment]
Ugh. So you mean we need to use a 1.18 vintage kubectl against a (for example) 1.16 cluster? That seems wrong...

Good detective work!

**Comment by mhyllander** [comment]
Interesting. As a side note, I have been testing to create AKS clusters with "az aks create", with AAD integration enabled. I have tried both approaches, setting everything up with [Azure CLI](https://docs.microsoft.com/en-us/azure/aks/azure-ad-integration-cli) and using the [AKS-managed Azure AD](https://docs.microsoft.com/en-us/azure/aks/managed-aad) (in preview).

When I use the managed approach, I get the same authorization error as reported here, both with k8s 1.16.9 and 1.17.5. The CLI approach works for me, using kubectl 1.16.9 with and 1.17.5 with the respective kubernetes versions.

**Comment by posix4e** [comment]
Cool , it is broken using dex as well

**Comment by DavidParks8** [comment]
looks like that particular code in k8s hasn't been touched in 2 years. I wonder if there could be a regression in the downstream oidc client. I'm looking at https://github.com/coreos/go-oidc/blob/395130aca3cb2a96cd195d9e9296d5a4351a1669/oidc.go#L113 right now.

**Comment by satishsudini** [comment]
Adding this -oidc-ca-file=/etc/kubernetes/certs/oidc-ca.crt resolved the issue

**Comment by DavidParks8** [comment]
adding that where? And does the file already exist? Where did you get it?

**Comment by satishsudini** [comment]
In the apiserver when we removed --oidc-client-id=spn argument we saw the cert error downloaded from MS and added --oidc-ca-file=/etc/kubernetes/certs/oidc-ca.crt

**Comment by DavidParks8** [comment]
You shouldn't need to do that. @satishsudini could you possible post more logs so we could see the reason the cert was not trusted in the first place? I'm wondering if there is some bug in intermediate cert validation somewhere.

**Comment by vinisoto** [comment]
@ericschung, @DavidParks8: I'm also facing  **(error: You must be logged in to the server (Unauthorized))** using kubectl 1.18, kubernetes 1.18 and aks-engine v0.52.0. Should I open a different issue for this?

**Comment by jikuma** [comment]
@satishsudini I checked in the container k8s_kube-apiserver cotnainer and can not find the cert /etc/kubernetes/certs/oidc-ca.crt

**Comment by ericsuhong** [comment]
@satishsudini Did you manually inject oidc-ca.crt file? I am still not sure why apiserver is not able to validate certificate from microsoft openid-configuration endpoint. It is a signed by a known DigiCert root cert.

**Comment by jackfrancis** [comment]
Can anyone still unable to solve this try with this new release of AKS Engine:

https://github.com/Azure/aks-engine/releases/tag/v0.55.2

The above includes a change that mounts the k8s certs directory in the apiserver container.

**Comment by jackfrancis** [comment]
This has been fixed w/ #3800

**Comment by jackfrancis** [comment]
This is the release you need to get the fix:

https://github.com/Azure/aks-engine/releases/tag/v0.55.3
